lib/widgets/update_expense.dart

```diff
@@ -1,98 +1,158 @@
+                autofocus: true,
+                autocorrect: false,
+                decoration: const InputDecoration(hintText: "Expense Name"),
+                textInputAction: TextInputAction.done,
+                validator: (name) {
+                  if (name != null && name.isNotEmpty) {
+                    return null;
+                  } else {
+                    return 'Enter a valid expense name';
+                  }
+                },
+              ),
+              const SizedBox(
+                height: 20.0,
+              ),
+              FutureBuilder<List<ExpenseCategory?>?>(
+                  future: widget.apiService.getExpenseCategories(),
+                  builder: (context, snapshot) {
+                    return DropdownButtonFormField<ExpenseCategory>(
+                      hint:
+                          Text(widget.expenseItem.expensecategory.categoryname),
+                      onChanged: (newValue) {
+                        setState(() {
+                          _selectedExpenseCategory = newValue!;
+                        });
+                      },
+                      validator: (selected) {
+                        if (selected == null) {
+                          return 'Select an expense item';
+                        }
+                        return null;
+                      },
+                      items: snapshot.data
+                          ?.map((ec) => DropdownMenuItem<ExpenseCategory>(
+                                value: ec,
+                                child: Text(ec!.categoryname),
+                              ))
+                          .toList(),
+                    );
+                  }),
+              const SizedBox(
+                height: 20.0,
+              ),
+            ]),
+          ),
         ),
       ),
       actions: [
         TextButton(
             child: const Text('Save'),
             onPressed: () async {
-              _updatedExpenseItem = expenseItem.copyWith(
-                  expensevalue: double.parse(_editExpenseValueController.text),
-                  expensename: _editExpenseNameController.text,
-                  expensecategory: _selectedExpenseCategory);
-
-              await apiService.updateExpense(_updatedExpenseItem);
+              final currentState = formGlobalKey.currentState;
+              if (currentState == null) {
+                return;
+              }
+              if (currentState.validate()) {
+                ExpenseItem _updatedExpenseItem = widget.expenseItem.copyWith(
+                    expensevalue:
+                        double.parse(_editExpenseValueController.text),
+                    expensename: _editExpenseNameController.text,
+                    expensecategory: _selectedExpenseCategory);
+                await widget.apiService.updateExpense(_updatedExpenseItem);
+                _editExpenseNameController.clear();
+                _editExpenseValueController.clear();
```

Comment:
Shouldn't need to be cleared. By popping, we dispose of this `State` instance and will create another one if we show the page again.